cpp/src/groupby/hash/groupby.cu

```diff
@@ -589,13 +557,33 @@ rmm::device_uvector<size_type> extract_populated_keys(Map map,
 std::unique_ptr<table> groupby(table_view const& keys,
                                host_span<aggregation_request const> requests,
                                cudf::detail::result_cache* cache,
-                               bool keys_have_nulls,
-                               null_policy include_null_keys,
+                               bool const keys_have_nulls,
+                               null_policy const include_null_keys,
                                rmm::cuda_stream_view stream,
                                rmm::mr::device_memory_resource* mr)
 {
-  auto d_keys_ptr = table_device_view::create(keys, stream);
-  auto map        = create_hash_map(*d_keys_ptr, keys_have_nulls, include_null_keys, stream);
+  auto const num_keys            = keys.num_rows();
+  auto const null_keys_are_equal = null_equality::EQUAL;
+  auto const has_null            = nullate::DYNAMIC{cudf::has_nested_nulls(keys)};
+
+  auto preprocessed_keys = cudf::experimental::row::hash::preprocessed_table::create(keys, stream);
+  auto const comparator  = cudf::experimental::row::equality::self_comparator{preprocessed_keys};
+  auto const row_hash    = cudf::experimental::row::hash::row_hasher{std::move(preprocessed_keys)};
+  auto const d_key_equal = comparator.device_comparator(has_null, null_keys_are_equal);
+  auto const d_row_hash  = row_hash.device_hasher(has_null);
```

Comment:
Will this be called for all data types? Is there any performance penalty when using such complex comparator/hasher for plain old data types?

Reply:
We are using new row comparator/hasher consistently for all data types. Let me double check the performmance impact for naive data types. Will report shortly.

Reply:
Updates: No performance regression on naive data types.